brownie/project/main.py

```diff
@@ -761,14 +764,40 @@ def _install_from_github(package_id: str) -> str:
     shutil.move(installed, install_path)
 
     try:
-        if not install_path.joinpath("contracts").exists():
-            raise Exception
-        new(str(install_path), ignore_existing=True)
+        brownie_config: Dict = {"project_structure": {}}
+
+        contract_paths = set(
+            i.relative_to(install_path).parts[0] for i in install_path.glob("**/*.sol")
+        )
+        contract_paths.update(
+            i.relative_to(install_path).parts[0] for i in install_path.glob("**/*.vy")
+        )
+        if not contract_paths:
+            raise InvalidPackage(f"{package_id} does not contain any .sol or .vy files")
+        if install_path.joinpath("contracts").is_dir():
+            brownie_config["project_structure"]["contracts"] = "contracts"
+        elif len(contract_paths) == 1:
+            brownie_config["project_structure"]["contracts"] = contract_paths.pop()
+        else:
+            raise InvalidPackage(
+                f"{package_id} has no `contracts/` subdirectory, and "
+                "multiple directories containing source files"
+            )
+
+        with install_path.joinpath("brownie-config.yaml").open("w") as fp:
+            yaml.dump(brownie_config, fp)
```

Comment:
This will overwrite the config file. Are you sure no other method during installation is interacting with the config file now or in the future?

Reply:
Oh hrmm..  this is a good point. I was assuming it doesn't exist because I was staring at npm packages for so long. If it _does_ exist, all of this logic is moot.